crates/editor/src/signature_help.rs

```diff
@@ -291,28 +327,171 @@ impl SignatureHelpState {
 }
 
 impl SignatureHelpPopover {
-    pub fn render(&mut self, max_size: Size<Pixels>, cx: &mut Context<Editor>) -> AnyElement {
-        div()
-            .id("signature_help_popover")
-            .elevation_2(cx)
-            .overflow_y_scroll()
+    pub fn render(
+        &mut self,
+        max_size: Size<Pixels>,
+        window: &mut Window,
+        cx: &mut Context<Editor>,
+    ) -> AnyElement {
+        let Some(signature) = self.signature.get(*self.current_signature.borrow()) else {
+            return div().into_any_element();
+        };
+        let label = signature
+            .label
+            .clone()
+            .when(signature.label.is_empty(), |_| "<No Parameters>".into());
+        let signature_count = self.signature.len();
+        let signature_label = div()
             .max_w(max_size.width)
-            .max_h(max_size.height)
-            .on_mouse_move(|_, _, cx| cx.stop_propagation())
-            .on_mouse_down(MouseButton::Left, |_, _, cx| cx.stop_propagation())
+            .px_2()
+            .py_0p5()
             .child(
-                div().px_2().py_0p5().child(
-                    StyledText::new(self.label.clone())
-                        .with_default_highlights(&self.style, self.highlights.iter().cloned()),
-                ),
+                StyledText::new(label)
+                    .with_default_highlights(&self.style, signature.highlights.iter().cloned()),
             )
+            .into_any_element();
+        let signature = div()
+            .id("signature_help_documentation")
+            .overflow_y_scroll()
+            .track_scroll(&self.scroll_handle)
+            .flex()
+            .flex_col()
+            .max_h(max_size.height)
+            .child(signature_label)
+            .when_some(signature.documentation.clone(), |this, description| {
+                this.children(vec![
+                    div().border_primary(cx).border_1().into_any_element(),
+                    div()
+                        .max_w(max_size.width)
+                        .px_2()
+                        .py_0p5()
+                        .child(
+                            MarkdownElement::new(description, hover_markdown_style(window, cx))
+                                .code_block_renderer(markdown::CodeBlockRenderer::Default {
+                                    copy_button: false,
+                                    border: false,
+                                })
+                                .on_url_click(open_markdown_url),
+                        )
+                        .into_any_element(),
+                ])
+            })
+            .into_any_element();
+        let controls = if self.signature.len() > 1 {
+            let prev_button = div().flex().flex_row().justify_center().child({
+                let current_signature = self.current_signature.clone();
+                Button::new("signature_help_prev_button", "▴")
```

Comment:
Nothing will tell the user which bindings are for these actions, can we do something like

https://github.com/zed-industries/zed/blob/889d90706816fc5ad98b43501e0520e53ffc0506/crates/agent/src/inline_prompt_editor.rs#L522-L535

instead?